self-core-impl/src/test/java/com/selfxdsd/core/SelfCoreTestCase.java

```diff
@@ -107,5 +111,33 @@ public void closesStorageOnClose() throws Exception {
         Mockito.verify(storage, Mockito.times(1)).close();
     }
 
+    /**
+     * Should verify an authenticated user.
+     */
+    @Test
+    public void authenticatedUser() {
+        final User user = Mockito.mock(User.class);
+        Mockito.when(user.username()).thenReturn("john");
+        Mockito.when(user.email()).thenReturn("john@john.com");
+        final Provider provider = Mockito.mock(Provider.class);
+        Mockito.when(user.provider()).thenReturn(provider);
+        final Projects all = Mockito.mock(Projects.class);
+        Mockito.when(user.projects()).thenReturn(all);
+        final Authenticated authenticated = new Authenticated(user, "tok3n");
+        final Provider prov = new Authenticated(user, "tok3n").provider();
 
+        MatcherAssert.assertThat(
+            authenticated.username(),
+            Matchers.equalTo("john")
+        );
+        MatcherAssert.assertThat(
+            authenticated.email(),
+            Matchers.equalTo("john@john.com")
+        );
+        Mockito.verify(provider).withToken(anyString());
```

Comment:
@ANadrowski This line verifies that ``provider.withToken(...)`` has been called once with any String. I think you can make it even more precise, like this:

```java
Mockito.verify(provider).withToken("tok3n");
```

This way you are sure that ``withToken`` has been called with the initial token.

Reply:
ok :)